### PR TITLE
Milestone 2

### PR DESCRIPTION
**Pull request by vzdrizhni**
- Added: App, Display, ButtonPanel, and Button components.
- Changes have been pushed to [Heroku](https://calculator-react-js-vzdrizhni.herokuapp.com/)  👈

### CODE REVIEW

**Review by fatymahmed — CHANGES_REQUESTED** [review]
## Hi @vzdrizhni  
You have done a great job on this milestone 👏👏
However, there are some things that we need to fix.
**Kindly check my comments below**

## Status: Changes Required ♻️
_Feel free to reach out to me on slack (Fatima) if you have any questions or need any help_

File: README.md [hunk 1]
```diff
@@ -1,68 +1,152 @@
...
+  * [Built With](#built-with)
+  * [Usage](#usage)
+  * [Automated Test](#automated-test)
+* [Contributors](#contributors)
+* [Acknowledgements](#acknowledgements)
+* [License](#license)
+
+<!-- ABOUT THE PROJECT -->
+## About The Project
+  > to be added
```
Comment (fatymahmed) [inline README.md]: - Add more information about the project here, even though it is still being developed, you already know that this is a calculator project see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)

File: src/components/App.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import ButtonPanel from './buttonPanel.component';
+
+class App extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      result: 0,
+    };
+  }
```
Comment (fatymahmed) [inline src/components/App.js]: - Why are you setting the result to `0` here?, remember that this should be a functional component as mentioned on Pathwright, see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/appFunctionalComponent.png), therefore you do not need the constructor part rather you should only render the `Display` and `ButtonPanel` components in a `div`

File: src/components/button.component.jsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+
+const Button = (props) => (
+    <div className='button-container'>
+        <div className='button-group'>{props.buttonGroup.map(button => (
+            <button type='button' key={button.toString()}>{button}</button>
+            ))}
+        </div>
+    </div>
```
Comment (fatymahmed) [inline src/components/button.component.jsx]: - This component should accept the `button name` as an incoming prop  [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/button/acceptProp.png)  
which should be of type `String` [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/button/validateAsString.png)

File: src/components/display.component.jsx [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+const Display = ({result}) => (
+    <div>
+        <h4>{result}</h4>
+    </div>
+)
+
```
Comment (fatymahmed) [inline src/components/display.component.jsx:9]: - You should set the type of the `result` prop as `String` here see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/validateAsString.png)
  and the default value which will be `'0'` see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/DefaultProp.png)

File: src/components/button.component.jsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+
+const Button = (props) => (
+        <button>{props.button}</button>
+)
+
+export default Button
```
Comment (Oghenebrume50) [inline src/components/button.component.jsx]: Use proptypes package to validate that the datatype of the prop being received here

File: src/components/display.component.jsx (hunk 4 above)
Comment (Oghenebrume50) [inline src/components/display.component.jsx]: Use proptypes package to validate that the datatype of the prop being received here

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED

Hi @vzdrizhni good job making changes as outlined by the previous reviewer, everything thing seems to be in order but you need to properly use PropTypes to validate the props being passed to different components. This is why you were advised to add propTypes to the project in the first milestone

**Please make changes and request for another review** :slightly_smiling_face: 

**RAPHAEL NORIODE** :mask: STAY_SAFE

**Comment by vzdrizhni** [comment]
Changes required by TSE made:
- Use proptypes package to validate that the datatype of the prop.
- Added default prop to Display component.

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED

Hi @vzdrizhni this looks good :+1: kudos on making changes as earlier requested please merge and move on **This is approved**


**RAPHAEL NORIODE** :mask: STAY_SAFE

**Pull request merged**